include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -696,6 +696,47 @@ __pattern_remove_if(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec,
     return {std::ranges::begin(__r) + __idx, std::ranges::begin(__r) + __n};
 }
 
+//------------------------------------------------------------------------
+// reverse
+//------------------------------------------------------------------------
+
+template <typename _BackendTag, typename _ExecutionPolicy, typename _R>
+void
+__pattern_reverse(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _R&& __r)
+{
+    const auto __n = std::ranges::size(__r);
+    if (__n <= 1)
+        return;
+
+    oneapi::dpl::__par_backend_hetero::__parallel_for(_BackendTag{}, std::forward<_ExecutionPolicy>(__exec),
+                                                      unseq_backend::__reverse_functor<decltype(__n)>{__n}, __n / 2,
+                                                      oneapi::dpl::__ranges::views::all(std::forward<_R>(__r)))
```

Comment:
do we need this `views::all` call?
My understanding is that this is just to support `sycl::buffer` as a range.  I don't believe we are attempting to accept a `sycl::buffer` as input for the (non-experimental) parallel range API.

Reply:
An additional note for the future reference and better understanding.

`oneapi::dpl::__ranges::views::all` and alike add some methods on the top of minimal requirements of a range (`begin()`, `end()`, and `size()` for a sized range). One of such methods is `empty()`, which is occasionally used.

I've checked that the new reverse and reverse_copy patterns do not rely on anything using these methods.
Thinking further, `oneapi::dpl::__ranges::views::all` should be removed everywhere for the regular ranges together with the use of these "additional" methods.